Feat 109  2 3 column views

bryanbraun opened the pull request:
# READY FOR REVIEW

## Summary
Add support for 2 and 3 column views by using Drupal's default "Grid" options, as shown below.

![image](https://user-images.githubusercontent.com/1256329/73279323-f4806880-41ba-11ea-9e28-65cae47f6b5d.png)


## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. Set up a view, using the "grid" format, test using 2 and 3 columns, and confirm that the content displays in the number of columns configured and scales down appropriately on smaller screen widths.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

Commit:
feat(109): new grid patterns for 2 and 3 column pattern ui

Commit:
feat(109): more work on three column

Commit:
wip: add view grid template that will implement our grid html and styles

Commit:
wip: refactor View grid template to use a pattern file for rendering

bryanbraun commented on docroot/themes/humsci/humsci_basic/templates/views/views-view-grid.html.twig:
```diff
@@ -0,0 +1,40 @@
+  {% endfor %}
+{% endfor %}
+
+{{ pattern('grid', {items: viewItems, columns: options.columns}) }}
```
@nrambeck, I see the `grid` UI pattern being used but not `grid_two_column` or `grid_three_column`. Are those UI patterns earlier iterations that can be deleted from this PR?

silverli replied:
yes, @bryanbraun we started yesterday thinking the two and three column ui patterns could be used, but it looks like just the Grid one is used one.

silverli replied:
I can make a commit with those changes

Commit:
refactor(109): delete unused test patterns

Commit:
refactor(109): use calc for width of columns

silverli commented on docroot/themes/humsci/humsci_basic/templates/views/views-view-grid.html.twig:
```diff
@@ -0,0 +1,40 @@
+#}
+{% set viewItems = [] %}
+{% if title %}
+  <h3>{{ title }}</h3>
```
should this be an h2?

silverli replied:
Bryan said core template uses h3 so this is fine

silverli reviewed (changes requested):
We need to remember to add the compiled css in when this PR is ready

Commit:
fix(109): linting

bryanbraun committed:
fix: get the column number setting working for 2 & 3 columns

bryanbraun commented:
I've added the compiled CSS and fixed an issue where the column number setting wasn't getting passed to the "grid" UI pattern. As far as I can tell, it's all working now.

silverli reviewed (approved):
Looks great. Good catch on the column
